### PR TITLE
Initialize REST API Cache

### PR DESCRIPTION
**Pull request by KartikSuthar**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -368,7 +390,12 @@ module.exports = function (grunt) {
 				command: 'git clone https://github.com/buddyboss/buddyboss-platform-api.git',
 				cwd: SOURCE_DIR,
 				stdout: false
-			}
+			},
+			rest_performance: {
+				command: 'git clone -b Features/73_Performance https://github.com/buddyboss/buddyboss-platform-api.git',
```
Comment (dipskakadiya) [inline Gruntfile.js:395]: @KartikSuthar  We should merge this `Features/73_Performance` into the main branch as we releasing. Let me know your opinion.

File: src/bp-performance/bp-performance-loader.php [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+/**
+ * Set up the bp-performance component.
+ *
+ * @since BuddyBoss 1.6.0
+ */
+function bp_setup_performance() {
+	buddypress()->performance = new BP_Performance_Component();
+}
+add_action( 'bp_setup_components', 'bp_setup_performance', 1 );
```
Comment (dipskakadiya) [inline src/bp-performance/bp-performance-loader.php:22]: @KartikSuthar can't we need to comment on this also as we don't want to show components

**Review by dipskakadiya — COMMENTED** [review]
@KartikSuthar Can you address my below comment before I approve it. Code seems fine, good to merge

File: Gruntfile.js (hunk 1 above)
Comment (KartikSuthar) [reply Gruntfile.js]: @dipskakadiya Done.

File: src/bp-performance/bp-performance-loader.php (hunk 2 above)
Comment (KartikSuthar) [reply src/bp-performance/bp-performance-loader.php:22]: @dipskakadiya We have not called components, So this file was not loaded at all.

**Review by dipskakadiya — APPROVED** [review]
@KartikSuthar Approved

**Review by dipskakadiya — DISMISSED** [review]

File: src/bp-performance/classes/integrations/class-bb-media-albums.php [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+				$event_groups,
+				array(
+					'unique_id'         => 'id',
+					'purge_deep_events' => array_keys( $purge_single_events ),
+				),
+				true
+			);
+
+			$this->cache_endpoint(
+				'buddyboss/v1/media/albums<id>',
```
Comment (mehulkaklotar) [inline src/bp-performance/classes/integrations/class-bb-media-albums.php:106]: @KartikSuthar please check this endpoint path. slash should be there before`<id>`  like this `'buddyboss/v1/media/albums/<id>'`

**Review by mehulkaklotar — CHANGES_REQUESTED** [review]
@KartikSuthar please do the requested changes and let me know. I will merge it.

File: src/bp-performance/classes/integrations/class-bb-media-albums.php (hunk 3 above)
Comment (KartikSuthar) [reply src/bp-performance/classes/integrations/class-bb-media-albums.php]: Done

**Pull request merged**